### PR TITLE
Planning: OpenSpace: Use grid dp map for heuristic in hybrid A*

### PR DESCRIPTION
**Pull request by JasonZhou404**

### CODE REVIEW

File: modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc [hunk 1]
```diff
@@ -101,32 +82,27 @@ bool HybridAStar::RSPCheck(
...
+    for (const common::math::LineSegment2d& linesegment :
+         obstacle_linesegments) {
+      if (node->GetBoundingBox(vehicle_param_).HasOverlap(linesegment)) {
         return false;
       }
     }
   }
   return true;
-}
+}  // namespace planning
```
Comment (Capri2014) [inline modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc]: remove this

Comment (JasonZhou404) [reply modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc]: remove what? the "// namespace planning"?

**Comment by jinghaomiao** [comment]
@JasonZhou404 Please resolve the conflict.

File: modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc (hunk 1 above)
Comment (JasonZhou404) [reply modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc]: Done

File: modules/planning/open_space/coarse_trajectory_generator/grid_search.cc [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+      }
+    }
+  }
+  ADEBUG << "explored node num is " << explored_node_num;
+  return true;
+}
+
+double GridSearch::CheckDpMap(const double& sx, const double& sy) {
+  double index = Node2d::CalcIndex(sx, sy, xy_grid_resolution_, XYbounds_);
+  if (dp_map_.find(index) != dp_map_.end()) {
```
Comment (xiaoxq) [inline modules/planning/open_space/coarse_trajectory_generator/grid_search.cc]: Does it really work? @JasonZhou404

**Comment by JasonZhou404** [comment]
Yeah, what's the matter?

Xiangquan Xiao <notifications@github.com> 于2019年3月19日周二 下午11:58写道：

**Comment by xiaoxq** [comment]
Float numbers cannot be used as key, as they cannot be compared as exactly
equal.

**Pull request merged**